types/proposal.go

```diff
@@ -15,43 +15,41 @@ var (
 )
 
 // Proposal defines a block proposal for the consensus.
-// It refers to the block only by its PartSetHeader.
+// It refers to the block by BlockID field.
 // It must be signed by the correct proposer for the given Height/Round
 // to be considered valid. It may depend on votes from a previous round,
-// a so-called Proof-of-Lock (POL) round, as noted in the POLRound and POLBlockID.
+// a so-called Proof-of-Lock (POL) round, as noted in the POLRound.
+// If POLRound >= 0, then BlockID corresponds to the block that is locked in POLRound.
 type Proposal struct {
-	Type             SignedMsgType
-	Height           int64         `json:"height"`
-	Round            int           `json:"round"`
-	POLRound         int           `json:"pol_round"` // -1 if null.
-	Timestamp        time.Time     `json:"timestamp"`
-	BlockPartsHeader PartSetHeader `json:"block_parts_header"`
-	POLBlockID       BlockID       `json:"pol_block_id"` // zero if null.
-	Signature        []byte        `json:"signature"`
+	Type      SignedMsgType
+	Height    int64     `json:"height"`
+	Round     int       `json:"round"`
+	BlockID   BlockID   `json:"block_id"`
+	POLRound  int       `json:"pol_round"` // -1 if null.
```

Comment:
Let's keep the order here aligned with the Canonical order, so POLRound should come before BlockID